Shared/Samples/Identify raster cell/IdentifyRasterCellView.swift

```diff
@@ -0,0 +1,252 @@
+            }
+        }
+        
+        /// Updates the text shown in the callout using the attributes and coordinates of a given raster cell.
+        /// - Parameter cell: The raster cell to create the text from.
+        private func updateCalloutText(using cell: RasterCell) {
+            // Create the attributes text using the attributes of the raster cell.
+            let attributes = cell.attributes.map { "\($0.key): \($0.value)" }.sorted(by: >).joined(separator: "\n")
+            
+            // Create the coordinate texts using the extent of the cell's geometry.
+            guard let extent = cell.geometry?.extent else {
+                calloutText = attributes
+                return
+            }
+            let xCoordinate = "X: \(extent.xMin.formatted(.rounded))"
+            let yCoordinate = "Y: \(extent.yMin.formatted(.rounded))"
+            
+            // Update the callout text.
+            calloutText = "\(attributes)\n\n\(xCoordinate)\n\(yCoordinate)"
+        }
+    }
+    
+    /// An enumeration used to throw an error customized with a string.
+    enum CustomError: LocalizedError {
+        case message(String)
+        
+        /// The text description of the error.
+        var errorDescription: String? {
+            if case .message(let string) = self {
+                return NSLocalizedString(string, comment: "The description of the error thrown.")
+            }
+            return nil
+        }
+    }
+}
+
+/// An extension to add an on ended closure to the on long press gesture of the map view.
+private extension MapView {
+    /// The current screen point of a gesture.
+    static var gestureScreenPoint = CGPoint()
+    /// The current map point of a gesture.
+    static var gestureMapPoint = Point(x: 0, y: 0)
+    
+    /// Sets a closure to perform when the map view recognizes a long press gesture.
+    /// - Parameters:
+    ///   - action: The closure to perform when a long press is recognized.
+    ///   - onEnded: The closure to perform when the long press ends.
+    /// - Returns: A new `View` object.
+    func onLongPressGesture(perform action: @escaping (CGPoint, Point) -> Void, onEnded: ((CGPoint, Point?) -> Void)? = nil) -> some View {
+        self
+            .onLongPressGesture { screenPoint, mapPoint in
+                Self.gestureScreenPoint = screenPoint
+                Self.gestureMapPoint = mapPoint
+                
+                action(screenPoint, mapPoint)
+            }
+            .gesture(
+                LongPressGesture()
+                    .simultaneously(with: DragGesture())
+                    .onEnded { value in
+                        guard let onEnded else { return }
+                        
+                        if value.first != nil {
+                            onEnded(Self.gestureScreenPoint, Self.gestureMapPoint)
+                        }
+                    }
+            )
+    }
+}
```

Comment:
Nice idea but could we just use [onDragGesture(shouldBegin:onChanged:onEnded:onCancelled:)](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/mapview/ondraggesture(shouldbegin:onchanged:onended:oncancelled:)) ?

Reply:
I don't think so. If the user long pressed but didn't drag, the callout placement would still be offset when the magnifier disappears and the gesture ends because the drag gesture would never be called.